pywbem_mock/_wbemconnection_mock.py

```diff
@@ -305,6 +314,11 @@ def __init__(self, default_namespace=DEFAULT_NAMESPACE,
         self._mofwbemconnection = _MockMOFWBEMConnection(self,
                                                          self.mainprovider)
 
+        # TODO: This is just prototype. The registry would be
+        # namespace, classname, providersubclassinstance.
+        # Should be in BaseRepository or DefaultInstanceProvider.
```

Comment:
I don't think the provider registry should be in BaseRepository or DefaultInstanceProvider. It belongs where the dispatching of provider calls is done, and that should be here in this class.

Reply:
Cannot do that because that means maintaining reference to WBEMConnection class in MainProvider/BaseProvider/DefaultInstanceProvider.  That causes python circular reference.
**DISCUSSION**  The choices are live with circular reference or create attribute with cimrepository in the provider classes.   There is even another issue here in that we have to do this for  MainProvider and DefaultInstanceProvider  and ProviderDispatcher.  I am beginning to think the circular reference is not so bad.

Reply:
The circular reference is not an actual problem at runtime, it is more a design flaw if one needs that.

Reply:
Covered the relevant parts of this comment with new comments. CLosing this one.